robot_sort/robot_sort.py

```diff
@@ -96,9 +96,37 @@ def sort(self):
         """
         Sort the robot's list.
         """
-        # Fill this out
-        pass
 
+        # Passes all tests but is fails rules due to using self._list and storing variables :( 
+        # Loop through length of list 
+        for x in range(0, len(self._list) -1):
```

Comment:
Even though, it breaks some rules, I still a passing solution and I'm glad you got it working! Here's the optimal solution. Rework your code with the solution in mind to correct the broken rules 

self.swap_item()
        self.set_light_on()
        
        while self.light_is_on():
            # move right and find the smallest item that has not been sorted
            while self.move_right():
                if self.compare_item() > 0:
                    self.swap_item()
            # if the robot is at the end of the list and all items are sorted, we're done
            if self.compare_item() is None and self.can_move_right() is False:
                # place the current item down in the list and turn light off to indicate the list is sorted
                self.swap_item()
                self.set_light_off()
                break
            else:
                # find the spot in the list that is holding None
                # place the item in that position
                # move right and swap
                while self.move_left():
                    if self.compare_item() is None:
                        self.swap_item()
                        self.move_right()
                        self.swap_item()
                        break